libs/utils/utils.js

```diff
@@ -620,10 +620,13 @@ function decorateFooterPromo(config) {
 }
 
 export function loadIms(onReadyFn) {
-  if (window.adobeIMS) return;
+  if (window.adobeIMS) {
```

Comment:
I'm not sure this is enough - IMS is not necessarily ready if `adobeIMS` exists on the page. It is ready if `window.adobeid.onReady` has been called at least once. So one way to make sure we don't call `onReadyFn` prematurely would be to set a flag as part of the `window.adobeid.onReady` function body and then check here if that flag has already been set.

Another approach would be to use some event listeners to check when IMS is ready, something like
```
window.addEventListener('onImsLibInstance', (data) => {
    if(data?.detail?.instance) {
        console.log('IMS instance exists, IMS ready methods can be executed');
    }
});
window.dispatchEvent(new window.CustomEvent('getImsLibInstance'));
```

You'd likely need to use a combination of the two - if a flag has already been set in `window.adobeid.onReady`, execute `onReadyFn`. Otherwise, add the event listener logic and execute `onReadyFn` when the IMS instance is available.

Reply:
Thanks for the details, I was wondering about that 👍

Reply:
As discussed on Slack, I'll add the code to the `global-navigation` to not bulk up the utils even further.

We can still move it over to the utils if more consumers use the loadIms & run into the same issues